portfolio/src/main/webapp/script.js

```diff
@@ -100,4 +100,67 @@ function submitComment() {
 function likeComment(commentId) {
   const request = new Request(`/like?commentId=${commentId}`, { method: 'POST' })
   fetch(request).then(() => getComments());
+}
+
+function checkLoginStatus() {
+  const request = new Request('/login', { method: 'GET' });
+  fetch(request)
+    .then(response => response.json())
+    .then(isLoggedIn => {
+      const deleteButtonContainer = document.getElementById('delete-container');
+      const leaveCommentForm = document.getElementById('leave-a-comment');
+      const loginContainer = document.getElementById('login-container');
+      if (isLoggedIn) {
+        buildLoggedInPage(deleteButtonContainer, leaveCommentForm, loginContainer)
```

Comment:
Its not just building the element it is also appending it. I also think its better to have method that creates the element and the caller appends to the parent element.
For the isLoggedIn case you can start with `deleteButtonContainer.append(createDeleteCommentsButton());` and then follow this pattern with the rest making necessary helper methods that create more complex groups of elements.

Reply:
Yap done! Big refactoring but it also allows me to reuse `createButton()` to create the thumbs up buttons.